sql/parser/types.go

```diff
@@ -107,10 +108,20 @@ func (node *DateType) String() string {
 
 // TimestampType represents a TIMESTAMP type.
 type TimestampType struct {
+	Name string
 }
 
 func (node *TimestampType) String() string {
-	return "TIMESTAMP"
+	return node.Name
+}
+
+// IntervalType represents a INTERVAL type
+type IntervalType struct {
+	Name string
```

Comment:
The presence of `DateType.Name` is to support a `TIME` type. We should just remove that support from the parser and then remove `DateType.Name`. I don't see a reason to add `Name` fields for `{Timestamp,Interval}Type`.